src/test/java/unit/BqDatasetNumTables.java

```diff
@@ -0,0 +1,159 @@
+package unit;
+
+import static harness.utils.ExternalBQDatasets.randomDatasetId;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+import bio.terra.cli.serialization.userfacing.resource.UFBqDataset;
+import com.google.api.services.bigquery.model.DatasetReference;
+import harness.TestCommand;
+import harness.TestUsers;
+import harness.baseclasses.SingleWorkspaceUnit;
+import harness.utils.Auth;
+import harness.utils.ExternalBQDatasets;
+import java.io.IOException;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+/** Tests for including the number of tables in a BQ dataset resource's description. */
+@Tag("unit")
+public class BqDatasetNumTables extends SingleWorkspaceUnit {
+  // external dataset to use for creating BQ dataset references in the workspace
+  private DatasetReference externalDataset;
+
+  // name of table in external dataset
+  private String externalDatasetTableName = "testTable";
+
+  @BeforeAll
+  @Override
+  protected void setupOnce() throws Exception {
+    super.setupOnce();
+    externalDataset = ExternalBQDatasets.createDataset();
+    ExternalBQDatasets.grantWriteAccess(
+        externalDataset, workspaceCreator.email, ExternalBQDatasets.IamMemberType.USER);
+
+    // grant the user's proxy group access to the dataset so that it will pass WSM's access check
+    // when adding it as a referenced resource
+    ExternalBQDatasets.grantWriteAccess(
```

Comment:
Do these need to be separate? I think the user should be included in their proxy group

Reply:
Good point! Removed separate access grant to user email.